### DISCUSSION TITLE
Uno App with plug-in support.

### DISCUSSION THREAD

**Discussion by DavidM29** [unoplatform/uno#18100] [Q&A]
Hello, I am creating an application which must be extensible, because it must be tailored for different customers differently. I am guessing that the app must be able to take plug-ins and these plug-in will be developed for every customer specifically.

This is the first time I am developing on Uno and the first time I am developing this kind of application so could someone point me in a direction that I should go?

Thanks in advance.

**Comment by jeromelaban** [comment]
Thanks for the question. We do not provide guidance for this type of architecture, but it generally depends on the level of customization that you need and this is a general development concern that is not specific to Uno. You can use build configurations, external assemblies loading, server side configuration, etc...

That being said, note that loading external custom assemblies at runtime may be restricted in some scenarios, particularly on iOS.

**Reply by DavidM29** [reply]
Thanks for the answer and sorry if the question was not exactly related to Uno.
I found some general articles for .NET which I decided to follow, I was just curious if there were some things that had to be done differently for Uno specifically.
